content/en/references/internal-endpoints.md

```diff
@@ -0,0 +1,35 @@
+---
+title: Internal Endpoints
+weight: 120
+tags:
+- internal-endpoints
+- localstack-endpoints
+description: >
+  Overview of LocalStack and AWS specific internal endpoints for local development and testing
+---
+
+LocalStack provides several internal endpoints for various local AWS services and LocalStack-specific features. These endpoints are not part of the official AWS API and are available in the `/_localstack` and `/_aws` paths. You can use [`cURL`](https://curl.se/) or your favourite HTTP REST client to access endpoints.
+
+## LocalStack endpoints
+
+The API path for the LocalStack internal resources is `/_localstack`. The following endpoints are available:
+
+| Endpoint                             | Description                                                                                               |
+| ------------------------------------ | --------------------------------------------------------------------------------------------------------- |
+| `/_localstack/health`                | Get a list of all the running AWS services in your LocalStack container                                   |
+| `/_localstack/plugins`               | Get a list of all the information about LocalStack's Plux plugins                                         |
+| `/_localstack/init`                  | Get the state of the initialization procedure after Init hooks setup                                      |
+| `/_localstack/cloudformation/deploy` | Deploy your CloudFormation templates from a local web user-interface                                      |
+| `/_localstack/diagnose`              | Get specific image hashes and compare them with the latest image along with sensitive network information |
```

Comment:
```suggestion
| Endpoint | Description |
| ---------------------------------| --------------------------------------------------------------------------- |
| `/_localstack/health`| To check the available and running AWS services in LocalStack. You can use the endpoint to restart the LocalStack services.  |
| `/_localstack/plugins` | Shows the [Plux plugins ](https://docs.localstack.cloud/contributing/concepts/#plugins)information in LocalStack |
| `/_localstack/init`| Shows the initialization status after setting up [Init hooks](https://docs.localstack.cloud/references/init-hooks/) |
| `/_localstack/cloudformation/deploy` | Enables you to deploy CloudFormation templates locally through a web interface. |
| `/_localstack/diagnose`| Reports extensive and sensitive data from LocalStack instance (requires `DEBUG=1` setting in the environment variable). |
```

This endpoint  `/_localstack/cloudformation/deploy`  is not working for me, getting 404. Should it still work?

Reply:
Maybe add a new section with the details on how to restart the services.

```bash
 curl -v --request POST --header "Content-Type: application/json"  --data '{"action":"restart"}' http://localhost:4566/_localstack/health

 curl -v --request POST --header "Content-Type: application/json"  --data '{"action":"kill"}' http://localhost:4566/_localstack/health
```

Perhaps we should also mention these endpoints?
```bash
POST /_localstack/state/<service>/save take a snapshot the given service
POST /_localstack/state/<service>/load load the most recent snapshot of the given service
POST /_localstack/state/reset
POST /_localstack/state/<service>/reset
```
https://docs.localstack.cloud/references/persistence-mechanism/#endpoints

One more that is working with `ENABLE_CONFIG_UPDATES=1` 
```bash
/_localstack/config
```
https://docs.localstack.cloud/references/configuration/#security
I am not sure how exactly it works based on the code.
https://github.com/localstack/localstack/blob/10f82fdcd1d56c5a5ac5756b9cd793f4a9294738/localstack/services/internal.py#L302

Reply:
Hi @danopia, yes, these endpoints are only available with the pro version.